cmd/clusterctl/phases/pivot.go

```diff
@@ -239,24 +243,28 @@ func moveClusterSecrets(from sourceClient, to targetClient, cluster *clusterv1.C
 		return err
 	}
 
+	toCluster, err := to.GetCluster(cluster.Name, cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
 	for _, secret := range secrets {
-		if err := moveSecret(from, to, secret); err != nil {
+		if err := moveSecret(from, to, secret, toCluster); err != nil {
 			return errors.Wrapf(err, "failed to move Secret %s/%s", secret.Namespace, secret.Name)
 		}
 	}
 	return nil
 }
 
-func moveSecret(from sourceClient, to targetClient, secret *corev1.Secret) error {
+func moveSecret(from sourceClient, to targetClient, secret *corev1.Secret, toCluster *clusterv1.Cluster) error {
 	klog.V(4).Infof("Moving secret %s/%s", secret.Namespace, secret.Name)
 
 	// New objects cannot have a specified resource version. Clear it out.
 	secret.SetResourceVersion("")
-
-	// remove the UID from ownerReferences as it will be different across clusters
-	for i := 0; i < len(secret.OwnerReferences); i++ {
-		secret.OwnerReferences[i].UID = ""
-	}
+	// Clear the owner reference since its UID is different per cluster
+	// and reset it based on the cluster UID of the target cluster
+	secret.SetOwnerReferences(nil)
```

Comment:
I don't believe this is required - doesn't the next line overwrite the entire field?

Reply:
Yeah good catch